### ISSUE TITLE
A term can appear in multiple synonym groups

### ISSUE THREAD

**Issue by ethbib-susanne** [klokantech/kartenportal-tickets#201]
J don't come along with the synonym list. J made different tests but didn't succeed. Maybe you can make an example following these instructions:

There is a very complecated subject in the ETH index that is called "HYDROGEOLOGISCHE KARTEN + HYDROLOGISCHE KARTEN + HYDROGRAPHISCHE KARTEN". In the swissbib catalog the subject also comes in Englisch and French "HYDROGEOLOGICAL MAPS + HYDROLOGICAL MAPS + HYDROGRAPHIC MAPS" and "CARTES HYDROGÉOLOGIQUES + CARTES HYDROLOGIQUES + CARTES HYDROGRAPHIQUES" but not in Italian. According to this subject you always harvest automatically all ETH subject in three languages (D, E, F). 

The goal is to unite the subject "Hydrologie"/"Hydrology"/"Hydrologie" with the subject "HYDROGEOLOGISCHE KARTEN + HYDROLOGISCHE KARTEN + HYDROGRAPHISCHE KARTEN". Only the terms "Hydrologie"/"Hydrology"/"Hydrologie" should appear ind the list of proposed subjects according to the chosen language but the search must use the term ""HYDROGEOLOGISCHE KARTEN + HYDROLOGISCHE KARTEN + HYDROGRAPHISCHE KARTEN" because there is no subject "Hydrologie" in the ETH index.

In this example we have an additional problem because also the subject "Hydrogeologie"/"Hydrogeology"/"Hydrogeologie" must be united with the search term ""HYDROGEOLOGISCHE KARTEN + HYDROLOGISCHE KARTEN + HYDROGRAPHISCHE KARTEN". Only the terms "Hydrogeologie"/"Hydrogeology"/"Hydrogeologie" Only "Hydrologie"/"Hydrology"/"Hydrologie" should appear ind the list of proposed subjects according to the chosen language but the search must use the term ""HYDROGEOLOGISCHE KARTEN + HYDROLOGISCHE KARTEN + HYDROGRAPHISCHE KARTEN".

Also the subject "Hydrographie"/"Hydrography"/"Hydrogaphie" ...

**Comment by klokan** [comment]
A single term can not be used in two different synonym groups in the existing implementation. It is not part of the specification. But we may change the implementation (@keosak). The updates will follow in this ticket.

Regarding the failure of your testing:
Please read carefully the email with instructions. The changes made in the online spreadsheet are reflected in the search system only after you visit the provided update link and at soonest THE NEXT DAY, because reindexing happens over night. It is impossible to make multiple tests in one day... there is always overnight delay before new groups of synonyms are applied.

Please do not delete the content of the spreadsheet completely again. I have restored it.

**Comment by klokan** [comment]
Additional implementation comments in https://github.com/klokan/maprank-server/issues/22

**Comment by klokan** [comment]
Fixed - but not yet visible in http://klokan.github.io/maprank-client/.
Test term "hydrogeology" does not run there yet. It should after reindexing - i.e. tomorrow.

**Comment by keosak** [comment]
Synonyms in the Google Spreadsheet can now be prefixed with a "~" character. It means that the term will be searched inside subjects instead of being matched directly against them. I have already split the "hydrology + hydrogeology + hydrography" term in the spreadsheet to demonstrate the feature.
